### ISSUE TITLE
[Map] Cannot select overlapping polygons

### ISSUE THREAD

**Issue by DaoyuT** [google/ground-platform#749]
When polygon A is inside polygon B. Clicking A will only select B.

**Comment by gino-m** [comment]
@jacobmclaws Any ideas on what disambiguation might look like when clicking overlapping polygons? On the Android app we display a modal dialog with a list of all features at the clicked point.

**Comment by gino-m** [comment]
@DaoyuT Can we do something link this for now?

![Screenshot_1624557797](https://user-images.githubusercontent.com/228050/123311253-f3a94e80-d4f4-11eb-8ca8-956fa31aa6e9.png)

**Comment by DaoyuT** [comment]
Sounds good, I would put:
1. feature id
2. layer name
3. a ground pin with the layer's color
4. type of the feature??? (Point/ polygon, I though this would be helpful, not sure)
in each list item in the dialog.

WDYT?

**Comment by gino-m** [comment]
Yes, all fo the above would be great. For a first pass, we can omit the pin/icon.

The real challenge is determining which polygons are under the click locations, since Maps API doesn't provide them to us. We had a similar problem in Android, and @scolsen converted GeoJSONs to loops and holes so that `containsLocation` could be used to determine which polygons fall under the target location. One approach is described here:

https://stackoverflow.com/questions/64778263/how-to-access-overlapping-geojson-polygons-to-display-in-infobox

**Comment by gino-m** [comment]
> Sounds good, I would put:
> 
> 1. feature id
> 2. layer name
> 3. a ground pin with the layer's color
> 4. type of the feature??? (Point/ polygon, I though this would be helpful, not sure)
>    in each list item in the dialog.
> 
> WDYT?

The label behavior is described in https://github.com/google/ground-platform/issues/344#issuecomment-857944740.  @parulraheja98 has a PR out for this, should be merged soon!
